File: elixir/medium/2971/2971.exs
defmodule Solution do
  @spec largest_perimeter(nums :: [integer]) :: integer
  def largest_perimeter(nums) do
    nums
    |> Enum.sort()
    |> then(&do_largest_perimeter(&1, 0, -1))
  end

  def do_largest_perimeter([], _acc, result), do: result

  def do_largest_perimeter([head | tail], acc, _result) when acc > head do
    do_largest_perimeter(tail, acc + head, acc + head)
  end

  def do_largest_perimeter([head | tail], acc, result) do
    do_largest_perimeter(tail, acc + head, result)
  end
end

ExUnit.start()

defmodule AssertionTest do
  use ExUnit.Case, async: true

  test "test largest_perimeter" do
    assert Solution.largest_perimeter([5, 5, 5]) == 15
    assert Solution.largest_perimeter([1, 12, 1, 2, 5, 50, 3]) == 12
    assert Solution.largest_perimeter([5, 5, 50]) == -1
  end
end

File: py/medium/2971/2971.py
from typing import List


class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        sorted_nums = sorted(nums)
        acc = 0
        result = -1
        for n in sorted_nums:
            if acc > n:
                result = acc + n
            acc += n
        return result


solution = Solution()
assert solution.largestPerimeter([1, 12, 1, 2, 5, 50, 3]) == 12
assert solution.largestPerimeter([5, 5, 5]) == 15
